runners/direct-java/build.gradle

```diff
@@ -93,6 +89,10 @@ dependencies {
   validatesRunner project(path: ":sdks:java:core", configuration: "shadowTest")
   validatesRunner project(path: project.path, configuration: "shadow")
   validatesRunner project(path: project.path, configuration: "shadowTest")
+  //runtime dependencies of the tests
+  permitUnusedDeclared library.java.vendored_grpc_1_26_0
+  permitUnusedDeclared project(":runners:java-fn-execution")
+  permitUnusedDeclared project(":sdks:java:fn-execution")
```

Comment:
This seems like a heavyweight approach. Could you elaborate on why these are permitted? Perhaps providing some evidence of the failures would help determine a more nuanced approach.

Reply:
The comment "runtime dependencies of the tests" indicates to me that the proper approach is to change these dependencies to a test-only configuration like `shadowTest`.

Reply:
Changing these dependencies to test-only along with shadowTest configuration still causes the problem:
```
* What went wrong:
Execution failed for task ':runners:direct-java:analyzeClassesDependencies'.
> Dependency analysis found issues.
  unusedDeclaredArtifacts: 
   - beam.runners:java-fn-execution:2.28.0-SNAPSHOT@jar
   - beam.sdks.java:fn-execution:2.28.0-SNAPSHOT@jar
   - org.apache.beam:beam-vendor-grpc-1_26_0:0.3@jar

```
These dependencies have to be UnusedDeclared.

Reply:
Oops, shadowTest still puts things on the compile classpath. So maybe `testRuntimeOnly` is the correct configuration. Anyway, it doesn't need to block this PR since there's no regression.